dot/parachain/availability-distribution/availability_distribution.go

```diff
@@ -271,7 +271,70 @@ func (ad *AvailabilityDistribution) ProcessBlockFinalizedSignal(_ parachaintypes
 func (ad *AvailabilityDistribution) processAvailabilityDistributionMessageFetchPoV(
 	msg parachaintypes.AvailabilityDistributionMessageFetchPoV,
 ) error {
-	return nil // TODO: implement #4489
+	rt, err := ad.blockState.GetRuntime(msg.RelayParent)
+	if err != nil {
+		return err
+	}
+
+	authorityID, err := ad.sessionCache.GetAuthorityID(msg.FromValidator, msg.RelayParent, rt)
+	if err != nil {
+		return err
+	}
+
+	request := messages.NewOutgoingRequest(
+		authorityID,
+		&messages.PoVFetchingRequest{
+			CandidateHash: msg.CandidateHash,
+		})
+
+	sendRequests := messages.SendRequests{
+		Requests:       []*messages.OutgoingRequest{request},
+		IfDisconnected: messages.ImmediateError,
+	}
+
+	ad.subSystemToOverseer <- sendRequests
+
+	result := <-request.Result
+	defer close(msg.PovCh)
```

Comment:
why not place the defer at the function start? so the earlier returns in case of error will also close the channel